rskj-core/src/main/java/co/rsk/scoring/PeerScoringInformation.java

```diff
@@ -45,10 +47,38 @@ public PeerScoringInformation(PeerScoring scoring, String id, String type) {
         this.id = id;
     }
 
+    @VisibleForTesting
+    public PeerScoringInformation(String type, boolean goodReputation,

```

Comment:
Could we consider the alternative of having an ScoringForTest class (defined in test code that extends Scoring and override the getter methods), to inject a Scoring with some injected values? In this way, we could diminish the production code related with testing

Reply:
well, the fact is that I think a class should only have one simple constructor and any other complex construction should be a static method. It's more transparent

Reply:
Well, you can do that, for static factory methods, that ARE INVOLVED in production use case. It is not the case with this one, AFAIK. It could be better to move all this to test code; the way I suggested in my previous comment removes any test-oriented-and-only-used-to-test code from production code

Reply:
there is a factory method: construct a PeerScoringInformation given a Scoring object. I agree we should remove test oriented stuff and we also need to use constructors in a more transparent way

Reply:
https://github.com/rsksmart/rskj/pull/1351/commits/fc09e75791cb9e36ce383ede3fe7d3e81bdf6515